test/leafTest.js

```diff
@@ -0,0 +1,151 @@
+          //       2B6eD29A95753C3Ad948348e3e7b1A251080Ffb9,000000000771D8DB,000CA009
+          //              address                             creationBlock     id
+          operator: "0x2B6eD29A95753C3Ad948348e3e7b1A251080Ffb9",
+        },
+      ]
+
+      testData.forEach(async (test) => {
+        const leaf = await leafInstance.operator(BigNumber.from(test.leaf))
+        expect(leaf).to.equal(test.operator)
+      })
+    })
+  })
+
+  describe("creationBlock", async () => {
+    it("retrieves the creation block from the leaf", async () => {
+      const testData = [
+        {
+          leaf: "0x4ddc2d193948926d02f9b1fe9e1daa0718270ed50000000000DFD871001995B5",
+          //       4ddc2d193948926d02f9b1fe9e1daa0718270ed5,0000000000DFD871,001995B5
+          //              address                             creationBlock     id
+          creationBlock: "0xDFD871",
+        },
+        {
+          leaf: "0x176f3dab24a159341c0509bb36b833e7fdd0a1320000000005E8EEE00008F573",
+          //       176f3dab24a159341c0509bb36b833e7fdd0a132,0000000005E8EEE0,0008F573
+          //              address                             creationBlock     id
+          creationBlock: "0x5E8EEE0",
+        },
+        {
+          leaf: "0x2b6ed29a95753c3ad948348e3e7b1a251080ffb9000000000771D8DB000CA009",
+          //       2b6ed29a95753c3ad948348e3e7b1a251080ffb9,000000000771D8DB,000CA009
+          //              address                             creationBlock     id
+          creationBlock: "0x771D8DB",
+        },
+      ]
+
+      testData.forEach(async (test) => {
+        const leaf = await leafInstance.creationBlock(BigNumber.from(test.leaf))
+        expect(leaf).to.equal(test.creationBlock)
+      })
+    })
+  })
+
+  describe("id", async () => {
+    it("retrives the operator's id from the leaf", async () => {
+      const testData = [
+        {
+          leaf: "0x4ddc2d193948926d02f9b1fe9e1daa0718270ed50000000000DFD871001995B5",
+          //       4ddc2d193948926d02f9b1fe9e1daa0718270ed5,0000000000DFD871,001995B5
+          //              address                             creationBlock     id
+          id: "0x1995B5",
+        },
+        {
+          leaf: "0x176f3dab24a159341c0509bb36b833e7fdd0a1320000000005E8EEE00008F573",
+          //       176f3dab24a159341c0509bb36b833e7fdd0a132,0000000005E8EEE0,0008F573
+          //              address                             creationBlock     id
+          id: "0x8F573",
+        },
+        {
+          leaf: "0x2b6ed29a95753c3ad948348e3e7b1a251080ffb9000000000771D8DB000CA009",
+          //       2b6ed29a95753c3ad948348e3e7b1a251080ffb9,000000000771D8DB,000CA009
+          //              address                             creationBlock     id
+          id: "0xCA009",
+        },
+      ]
+
+      testData.forEach(async (test) => {
+        const leaf = await leafInstance.creationBlock(BigNumber.from(test.leaf))
```

Comment:
Shouldn't this be querying the ID and not the creation block?

Reply:
yes - and now i'm extremely concerned about why the test was passing

Reply:
okay, turns out the assertions don't run in a `forEach`, so i'm doing a good ol' fashioned for loop

Reply:
I wonder if I was bumping up into a race condition @_@